src/leaflet.filelayer.js

```diff
@@ -84,7 +87,11 @@
                 var layer;
                 try {
                     this.fire('data:loading', { filename: file.name, format: parser.ext });
-                    layer = parser.processor.call(this, e.target.result, parser.ext);
+                    var result = e.target.result;
+                    if (parser.decompressor) {
+                        result = parser.decompressor(result);
+                    }
```

Comment:
couldn't we include the decompression in the `parser.processor` function if a decompressor is defined?